text/0000-private_data_purge.md

```diff
@@ -0,0 +1,179 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+To enable private data purge feature, we introduce a new API in chaincode shim. Below is the function signature of this new API.
+
+**`PurgePrivateData(collection, key string) error`**
+
+As far as the latest state of a channel, maintained in statedb, is concerned, this new operation, `PurgePrivateData`, is no different from existing operation, `DeletePrivateData`. Hence, an application or chaincode would not notice any difference in behavior. The additional semantics associated with `PurgePrivateData` primarily impact private data store and reconciliation of private data. Both of these are not exposed to end users as of now (beyond the APIs mentioned above).
+
+
+For understanding the usage of this API, lets assume a scenario where a chaincode defines a private data collection and the collection contains personally identifiable information for customers such that the chaincode maintains PII info of a customer in the form of three keys, namely {CustomerID}_PII_Name, {CustomerID}_PII_DOB, and {CustomerID}_PII_Address. While name and date-of-birth may not change, a customer may have updated his address a few times and hence peers of eligible orgs would have persisted all the historical values for the address. Now, the customer may want the Fabric network to purge any of its PII data.
+
+In order to achieve this, the application may invoke a chaincode transaction in which the chaincode in turn invokes the *PurgePrivateData* API three times - once for each of the above mentioned PII keys. Finally, the application would get enough endorsements to satisfy the endorsement policies for each of the above keys and submits the transaction to the orderers. Once the transaction is committed on a peer, the peer stops sending out any version of any of these private data keys and eventually deletes from the persistent stores.
+
+Note that we will gate this new API behind a capability tag so that this new operation is interpreted on all the peers in a channel and assures the user that the data will be purged if the transaction is committed as a valid transaction. In the absence of the capability, the effect of the API will be exactly same as invoking the API `DeletePrivateData`.
```

Comment:
If the private data purge capability is not enabled for a channel, calls to `PurgePrivateData` should fail.

Reply:
Yes, we can very well return error during endorsement however, irrespective of this, the important point is to define the behavior at commit path. During the commit, as mentioned above in line 80, if we don't treat this as a delete, the older version of the peer would cause `panic`.

Reply:
@jt-nti - I gave it a bit more thoughts and we can indeed gate this at the time of endorsement only as it's an entirely new operation. I have updated the text - Thanks for the feedback.